src/Sets/Zonotope.jl

```diff
@@ -493,6 +493,65 @@ function quadratic_map(Q::Vector{MT}, Z::Zonotope{N}) where {N, MT<:AbstractMatr
     return Zonotope(d, remove_zero_columns(h))
 end
 
+"""
+    bound_intersect_2D(Z::Zonotope, L::Line2D)
+
+Return the support function in the direction [0, 1] of the intersection between
+the given zonotope and line.
+
+### Input
+
+- `Z` -- zonotope
+- `L` -- vertical line 2D
+
+### Output
+
+The support function in the direction [0, 1] of the intersection between the
+given zonotope and line.
+
+### Notes
+
+The given line must be vertical and the algorithm assumes that the intetrsection
+between the given sets is not empty.
+
+### Algorithm
+
+This function implements [Algorithm 8.2, 1].
+
+[1] *Colas Le Guernic. Reachability Analysis of Hybrid Systems with Linear
+Continuous Dynamics. Computer Science [cs]. Université Joseph-Fourier - Grenoble
+I, 2009. English. fftel-00422569v2f*
+"""
+function bound_intersect_2D(Z::Zonotope, L::Line2D)
+    dim(X) == 2 || throw(ArgumentError("the dimension of the set, but needs" *
+                                       " to be 2, but is $(dim(X))"))
```

Comment:
we should probably check that the intersection of `Z` and `L` is non-empty since it is not expensive. (or use an argument to avoid that check)